### DISCUSSION TITLE
Feature request: support for dynamic choices

### DISCUSSION THREAD

**Discussion by scott2b** [piccolo-orm/piccolo#1250] [Ideas]
Since choices takes an enum rather than a callable, it is not clear how one would provide choices dynamically. It would be ideal if we could provide dynamic choices so that, for example, the choices themselves could be stored in a table.

**Comment by cheesycod** [comment]
Hmmm... how does Django and other ORMs handle this

**Comment by dantownsend** [comment]
We could let the user pass in a function.

It does introduce some challenges though. For example, we have to make sure Piccolo Admin keeps refetching the table schema just in case the choices have changed.

Also, would the function return a list of potential values, or an Enum.

**Comment by cheesycod** [comment]
maybe a dict of key: value's
